iwyu_include_picker.cc

```diff
@@ -1436,10 +1436,15 @@ void IncludePicker::ExpandRegexes() {
   for (const auto& incmap : quoted_includes_to_quoted_includers_) {
     const string& hdr = incmap.first;
     for (const string& regex_key : filepath_include_map_regex_keys) {
+      const string regex = regex_key.substr(1);
       const vector<MappedInclude>& map_to = filepath_include_map_[regex_key];
-      if (RegexMatch(regex_dialect, hdr, regex_key.substr(1)) &&
+      if (RegexMatch(regex_dialect, hdr, regex) &&
           !ContainsQuotedInclude(map_to, hdr)) {
-        Extend(&filepath_include_map_[hdr], filepath_include_map_[regex_key]);
+        for (const MappedInclude& target : map_to) {
+          filepath_include_map_[hdr].push_back(
+            MappedInclude(RegexReplace(regex_dialect, hdr, regex,
```

Comment:
Could you pull the `RegexReplace` result out into a temporary to keep line length down?

Ah, and I think the only misformat is that we usually indent continued lines by four spaces relative to previous line's indentation. Clang-format should fix that for you automatically.

Reply:
Clang-format made it fit on two lines, removing the need for a local variable. Using local variables isn't always useful or desired, as described in [abseil.io/tips/161](https://abseil.io/tips/161).

Reply:
Yeah, all else equal I prefer to avoid temporary locals, but long lines or lines with complex wrapping are usually harder for me to read, so I tend to err on the side of less complex expressions. But I think it looks good after the re-format, thanks!